tests/test_env.py

```diff
@@ -232,20 +232,57 @@ def test_process_env_install_from_index(
+@pytest.mark.parametrize_any("use_system_config", [True, False])
+def test_process_env_install_from_index_not_found_env_var(
+    tmpvenv_active: tuple[py.path.local, py.path.local],
+    monkeypatch,
+    create_local_package_index,
+    use_extra_indexes,
+    use_extra_indexes_env,
+    use_system_config,
 ) -> None:
     """
-    Attempt to install a package that does not exist from a pip index. Assert the appropriate error is raised.
+    Attempt to install a package that does not exist from a pip indexes defined in the env vars, in the pip config or in both.
+    This if the system config are used or not.
+    Assert the appropriate error is raised.
     """
-    with pytest.raises(env.PackageNotFound, match="Packages this-package-does-not-exist were not found in the given indexes."):
-        # pass use_system_config=False for security reasons (anyone could publish this package to PyPi)
-        env.process_env.install_for_config(
-            [Requirement.parse("this-package-does-not-exist")],
-            config=PipConfig(
-                index_url=local_module_package_index,
-            ),
+    with tempfile.TemporaryDirectory() as tmpdir:
+        index_url = create_local_package_index(tmpdir)
+
+        extra_index_urls = []
+        if use_extra_indexes:
+            extra_index_1 = create_local_package_index(tmpdir, "extra1")
+            extra_index_2 = create_local_package_index(tmpdir, "extra2")
+            extra_index_urls = [extra_index_1, extra_index_2]
+
+        extra_index_urls_env = ""
+        if use_extra_indexes_env:
+            extra_index_env1 = create_local_package_index(tmpdir, "extra_env1")
+            extra_index_env2 = create_local_package_index(tmpdir, "extra_env2")
+            extra_index_urls_env = f"{extra_index_env1} {extra_index_env2}"
+
+        monkeypatch.setenv("PIP_EXTRA_INDEX_URL", extra_index_urls_env)
+
+        expected_indexes = index_url
+        if use_system_config:
+            env_index_url_env = ", " + extra_index_urls_env.replace(" ", ", ") if use_extra_indexes_env else ""
+            # PIP_EXTRA_INDEX_URL expects the different urls to be given as a string with a single space separating them,
+            # while the output has commas separating the different values.
+            expected_indexes = expected_indexes + env_index_url_env
+
+        expected_indexes = expected_indexes + ", " + ", ".join(extra_index_urls) if extra_index_urls else expected_indexes
+
+        expected: str = (
+            "Packages this-package-does-not-exist were not "
+            "found in the given indexes. (Looking in indexes: %s)" % expected_indexes
         )
+        with pytest.raises(env.PackageNotFound, match=re.escape(expected)):
+            env.process_env.install_for_config(
+                [Requirement.parse("this-package-does-not-exist")],
+                config=PipConfig(index_url=index_url, extra_index_url=extra_index_urls, use_system_config=use_system_config),
```

Comment:
This test case never sets `index_url` to `None`. We should also test that scenario.